src/app/vendors/reducer.ts

```diff
@@ -38,7 +40,37 @@ export const vendors: Reducer<VendorsState, VendorsAction | AccountsAction> = (
           ...state.vendorsByCharacter,
           [characterId]: {
             vendorsResponse: vendorsResponse,
-            lastLoaded: new Date(),
+            lastLoaded: Date.now(),
+            error: undefined,
+          },
+        },
+      };
+    }
+
+    // augments the storedoverall all-vendors response,
+    // by inserting the item components from a single-vendor api response
+    case getType(actions.loadedSingle): {
+      const { characterId, vendorResponse, vendorHash } = action.payload;
+      const { vendorsByCharacter } = state;
+      const oldVendorsResponse = vendorsByCharacter[characterId]?.vendorsResponse;
+
+      // nothing about state needs changing if we didn't get back components
+      // (maybe sockets/etc are disabled at bnet right now?)
+      if (_.isEmpty(vendorResponse.itemComponents) || !oldVendorsResponse) {
+        return state;
+      }
+
+      const newItemComponents = vendorResponse.itemComponents; // big combined set, keyed by vendor hash
+      const existingItemComponents = oldVendorsResponse?.itemComponents; // set for a single vendor
```

Comment:
Comments seem backwards?

Reply:
yeah! unsure when that swapped. combination went through several iterations. thank you